types/polyfill-library/index.d.ts

```diff
@@ -0,0 +1,138 @@
+// Type definitions for polyfill-library 3.108
+// Project: https://github.com/financial-times/polyfill-library#readme
+// Definitions by: Qingrong Ke <https://github.com/keqingrong>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { Readable } from 'stream';
+
+export interface PolyfillConfig {
+    aliases: string[];
+    dependencies: string[];
+    spec: string;
+    repo: string;
+    docs: string;
+    license: string;
+    notes: string[];
+    browsers: { [browser: string]: string };
+    install: { [key: string]: string | string[] };
+}
+
+export interface PolyfillMeta extends PolyfillConfig {
+    detectSource: string;
+    baseDir: string;
+    hasTests: boolean;
+    isTestable: boolean;
+    isPublic: boolean;
+    size: number;
+}
+
+export interface Options {
+    /**
+     * Name of the JavaScript function to call after the polyfill bundle is loaded.
+     * @default false
+     */
+    callback?: false | string;
+    /**
+     * Whether to return the minified or raw implementation of the polyfills.
+     * @default true
+     */
+    minify?: boolean;
+    /**
+     * Whether to return all polyfills or no polyfills if the user-agent is unknown or unsupported.
+     * @default 'polyfill'
+     */
+    unknown?: 'ignore' | 'polyfill';
+    /**
+     * Which features should be returned if the user-agent does not support them natively.
+     * @default {}
+     */
+    features?: { [featureName: string]: { flags: Flag[] | Flag[] } };
```

Comment:
this (union) looks like simple mistake.

Btw this looks to be set or iterable in usage:
```ts
interface Options {
    features: {
        [featureName: string]: {
            flags: FeatureType<Flag>
        }
    }
}
type Flag = 'always' | 'gated';
type FeatureType<T extends Flag = Flag> =  Iterable<T> | Set<T>;
declare const options: Options;

// Normalise flags
for (const k of Object.keys(options.features)) {
    if (!options.features[k].flags) {
        options.features[k].flags = new Set();
    } else if (options.features[k].flags.constructor !== Set) {
        options.features[k].flags = new Set(options.features[k].flags);
    }
}
```

Reply:
Yes this is a clerical error. It was supposed to be like this.

```ts
features?: { [featureName: string]: { flags: Flag[] | Set<Flag> } };
```